daemon/main.c

```diff
@@ -31,8 +31,10 @@ void netdata_cleanup_and_exit(int ret) {
     analytics_free_data();
 
     char agent_crash_file[FILENAME_MAX + 1];
+    char agent_incomplete_shutdown_file[FILENAME_MAX + 1];
     snprintfz(agent_crash_file, FILENAME_MAX, "%s/.agent_crash", netdata_configured_varlib_dir);
-    (void) unlink(agent_crash_file);
+    snprintfz(agent_incomplete_shutdown_file, FILENAME_MAX, "%s/.agent_incomplete_shutdown", netdata_configured_varlib_dir);
```

Comment:
I am not sure if the name is proper for what is happening :thinking:  . I sent a `SIGTERM` to finish the agent, and everything described on this PR happened, but a `SIGTERM  is not something expected to generate an incomplete shutdown. My suggestion is we rename the file. If I misunderstood the main goal of the PR, please, can you explain why a normal end means an "incomplete shutdown"? Thanks.

Reply:
My understanding is that the goal here is to detect cases where the agent was asked to shut down gracefully, but is then not given sufficient time to do so. ‘incomplete shutdown’ sounds like a correct description to me provided that that situation is what we are actually checking for.

Reply:
I got your point @Ferroin , but considering the fact that the agent was running,  my understanding is the fact that an incomplete shutdown happens only when the shutdown process starts, but somehow it cannot go until the end. This was the motive I required to rename the file.

Reply:
The `incomplete` shutdown will happen if a SIGTERM is received and the agent fails (crashes or externally killed before it gracefully exits). In the happy path that it exits gracefully it will remove the file so it will not be detected upon restart.

We need to further improve this in subsequent PRs to measure the time it takes to shutdown etc

Reply:
I think what Thiago is saying, is that you can keep the logic the same, but change the filename to perhaps `.agent_shutting_down`...

Reply:
@MrZammler is right, I do not think the name is the best option to describe what is happening right now.
Considering that we will have another PRs to improve this, I will approve this PR for we do not postpone more the feature. Please, think about the possibility to rename the file.